File: bench/cases/closure-pipeline-1m/main.go
package main

import "fmt"

func add1(x int64) int64 {
	return x + 1
}

func reduce(xs []int64, f func(int64) int64) int64 {
	var sum int64 = 0
	for i := 0; i < len(xs); i++ {
		sum = sum + f(xs[i])
	}
	return sum
}

func main() {
	xs := make([]int64, 0)
	for i := int64(0); i < 10000000; i++ {
		xs = append(xs, i)
	}
	fmt.Println(reduce(xs, add1))
}

File: bench/cases/closure-pipeline-1m/main.ts
function add1(x: number): number {
  return x + 1
}

function reduce(xs: number[], f: (n: number) => number): number {
  let sum = 0
  for (let i = 0; i < xs.length; i = i + 1) {
    sum = sum + f(xs[i])
  }
  return sum
}

const xs: number[] = []
for (let i = 0; i < 10000000; i = i + 1) {
  xs.push(i)
}
console.log(reduce(xs, add1))
